### PR TITLE
Update apple_bundle name

### PR DESCRIPTION
**Pull request by shepting**
This makes Buck commands a little more idiomatic. e.g. `buck build //App:ExampleApp`

 * Update the `apple_bundle` name to be `ExampleApp`
 * Update the `apple_library` to be named `ExampleAppLibrar`
 * Add Apple Watch, and other CI checks.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -16,15 +16,21 @@ update_cocoapods:
 	pod install
 
 build:
-	$(BUCK) build //App:ExampleAppBundle
+	$(BUCK) build //App:ExampleApp
+
+watch:
```
Comment (dfed) [inline Makefile]: 👏

File: Makefile [hunk 2]
```diff
@@ -1,5 +1,5 @@
 
-.PHONY : log install_buck build targets pods audit debug test xcode_tests clean project audit
+.PHONY : log install_buck build watch message targets pods audit debug test xcode_tests clean project audit
```
Comment (bachand) [inline Makefile]: What does this line do? I’ve always wondered

File: Makefile [hunk 3]
```diff
@@ -44,11 +50,13 @@ audit:
 	$(BUCK) audit rules Pods/BUCK > Config/Gen/Pods-BUCK.py
 
 clean:
```
Comment (bachand) [inline Makefile]: Should we have `clean` depend on `killall`

File: Makefile (hunk 2 above)
Comment (shepting) [reply Makefile]: Make has trouble knowing when it should run a task again. Something about looking at file timestamps...

This seems to make it _always_ run that task.

File: Makefile [hunk 4]
```diff
@@ -16,15 +16,21 @@ update_cocoapods:
...
 
 debug:
-	$(BUCK) install //App:ExampleAppBundle --run --simulator-name 'Phone: iPhone XS'
+	$(BUCK) install //App:ExampleApp --run --simulator-name 'Phone: iPhone XS'
 
 targets:
 	$(BUCK) targets //...
 
-ci: install_buck targets build test project xcode_tests
+ci: install_buck targets build test project xcode_tests watch message
```
Comment (shepting) [inline Makefile]: This line runs extra builds on CI.

File: Makefile (hunk 3 above)
Comment (shepting) [reply Makefile]: I seem to remember being frustrated that some task killed my Xcode. I think that's why I made this change.

File: Makefile (hunk 2 above)
Comment (bachand) [reply Makefile]: 👍

**Pull request merged**